api/envoy/config/response_map/v3/response_map.proto

```diff
@@ -0,0 +1,89 @@
+syntax = "proto3";
+
+package envoy.config.response_map.v3;
+
+import "envoy/config/accesslog/v3/accesslog.proto";
+import "envoy/config/core/v3/base.proto";
+import "envoy/config/core/v3/substitution_format_string.proto";
+
+import "google/protobuf/struct.proto";
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.response_map.v3";
+option java_outer_classname = "ResponseMapProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: ResponseMap]
+// Response map filter :ref:`configuration overview <config_response_map>`.
+
+// The configuration to filter and change HTTP responses.
+// [#next-free-field: 6]
+message ResponseMapper {
+  // Filter to determine if this mapper should apply.
+  accesslog.v3.AccessLogFilter filter = 1 [(validate.rules).message = {required: true}];
+
+  // The new response status code if specified.
+  google.protobuf.UInt32Value status_code = 2 [(validate.rules).uint32 = {lt: 600 gte: 200}];
+
+  // The new local reply text if specified. It will be used in the `%LOCAL_REPLY_BODY%`
+  // command operator in the `body_format`.
+  core.v3.DataSource body = 3;
+
+  // A per mapper `body_format` to use when this mapper is matched.
+  core.v3.SubstitutionFormatString body_format_override = 4;
+
+  // HTTP headers to add to a matched response. This allows the response mapper to append, to add
+  // or to override headers of any matched response before it is sent to a downstream client.
+  repeated core.v3.HeaderValueOption headers_to_add = 5
+      [(validate.rules).repeated = {max_items: 1000}];
+}
+
+// The configuration to customize HTTP responses read by Envoy.
+message ResponseMap {
```

Comment:
This is same as LocalReplyConfig in HCM as well?

Reply:
Yep. From the comment above, o intend for there to be a common proto shared across use cases and that we should (eventually) migrate HCM to use this common type. I would appreciate thoughts/guidance on a migration strategy.